app/src/main/java/mainburg/planetenweg/LocationMarker.java

```diff
@@ -0,0 +1,146 @@
+ * updates it after a certain interval.
+ * Created by Markus on 04.10.2017.
+ */
+public class LocationMarker implements LocationListener {
+
+    private GoogleMap map;
+    private final Activity activity;
+    private final LocationManager gps;
+    private Marker position;
+    static final int GPS_PERMISSION_REQUEST_CODE = 1227;
+
+    private boolean active;
+
+    /**
+     * This variable indicates whether or not a location has been founb.
+     * It is needed to notify the user when his location could not have been determined.
+     */
+    private boolean locationFound;
+
+    public LocationMarker(Activity activity) {
+        this.map = map;
+        this.activity = activity;
+        active = false;
+        gps = (LocationManager) activity.getSystemService(Context.LOCATION_SERVICE);
+    }
+
+    /**
+     * Enables the Location Marker
+     * @param map The GoogleMap on which to place the marker
+     */
+    public void enable(GoogleMap map) {
+        this.map = map;
+        active = true;
+        if (ContextCompat.checkSelfPermission(activity, android.Manifest.permission.ACCESS_FINE_LOCATION)
+                == PackageManager.PERMISSION_GRANTED) {
+            refresh();
+        } else {
+            ActivityCompat.requestPermissions(activity, new String[] {
+                    android.Manifest.permission.ACCESS_FINE_LOCATION
+            },GPS_PERMISSION_REQUEST_CODE);
+        }
+    }
+    /**
+     * Disables the Location Marker
+     */
+    public void disable() {
+        position.remove();
+        active = false;
+    }
+
+    public boolean isEnabled() {
+        return active;
+    }
+
+    /**
+     * Sends a new request to determine the user's location.
+     * Make sure you have the required GPS permissions before calling this method.
+     */
+    public void refresh() {
+        gps.requestLocationUpdates(
+                LocationManager.GPS_PROVIDER,
+                5000, //the minimum time interval for notifications, in milliseconds. This field is only used as a hint
+                //to conserve power, and actual time between location, updates may be greater or lesser than this value. (See https://androidexample.com/GPS_Basic__-__Android_Example/index.php?view=article_discription&aid=68&aaid=93)
+                5,//the minimum distance interval for notifications
+                this);//The listener to which to send the updates
+
+        locationFound = false;
+        Timer timer = new Timer();
+        timer.schedule(new TimerTask() {
+            @Override
+            public void run() {
+                if (!locationFound) {
+                    Toast.makeText(activity, "Position konnte nicht ermittelt werden.", Toast.LENGTH_LONG);
+                }
+            }
+        }, 10000);
+
+    }
+
+    /************* Called after each 10 sec **********/
+    @Override
+    public void onLocationChanged(Location location) {
+        if (!active) {
+            return;
```

Comment:
Please do some logging here

Reply:
Log what?
I guess your problem is that you wouldn't want this method to be called when the LocationMarker is disabled. However, I couldn't find a way to unregister it from automatic updates.
That's why I use a boolean variable to decide whether to place the marker or not.

Reply:
Right, but we should log an error because this is a situation we wouldn't expect; a logged error gives us the information that something went wrong.